src/Illuminate/Validation/Rules/ArrayRule.php

```diff
@@ -28,25 +30,136 @@ public function __construct($keys = null)
             $keys = $keys->toArray();
         }
 
-        $this->keys = is_array($keys) ? $keys : func_get_args();
+        $keys = is_array($keys) ? $keys : func_get_args();
+
+        if (empty($keys)) {
+            return $this->addRule('array');
+        }
+
+        $keys = array_map(
+            static fn ($key) => enum_value($key),
+            $keys,
+        );
```

Comment:
You can simplify this to a first-class callable:
```suggestion
        $keys = array_map(enum_value(..), $keys);
```